.github/workflows/nx-integration-tests.yml

```diff
@@ -0,0 +1,186 @@
+        run: |
+          echo "input was"
+          echo "repository: ${repository}"
+          echo "pr_head: ${pr_head}"
+          echo "pr_base: ${pr_base}"
+
+  matrix-calculator:
+    outputs:
+      php_versions: ${{ steps.calculate-matrix.outputs.php_versions }}
+      database_versions: ${{ steps.calculate-matrix.outputs.database_versions }}
+      search_versions: ${{ steps.calculate-matrix.outputs.search_versions }}
+      message_queue_versions: ${{ steps.calculate-matrix.outputs.message_queue_versions }}
+      cache_versions: ${{ steps.calculate-matrix.outputs.cache_versions }}
+      http_cache_versions: ${{ steps.calculate-matrix.outputs.http_cache_versions }}
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Run Matrix Calulator
+        id: calculate-matrix
+        uses:  mage-os/github-actions/supported-services-matrix-calculator@main
+        with:
+          repository: ${{ inputs.repository }}
+          ref: ${{ inputs.pr_head }}
+
+      - name: Calculated Result
+        shell: bash
+        env:
+          php_versions: ${{ steps.calculate-matrix.outputs.php_versions }}
+          database_versions: ${{ steps.calculate-matrix.outputs.database_versions }}
+          search_versions: ${{ steps.calculate-matrix.outputs.search_versions }}
+          message_queue_versions: ${{ steps.calculate-matrix.outputs.message_queue_versions }}
+          cache_versions: ${{ steps.calculate-matrix.outputs.cache_versions }}
+          http_cache_versions: ${{ steps.calculate-matrix.outputs.http_cache_versions }}
+        run: |
+          echo "PHP Versions: $php_versions"
+          echo "database_versions: $database_versions"
+          echo "search_versions: $search_versions"
+          echo "message_queue_versions: $message_queue_versions"
+          echo "cache_versions: $cache_versions"
+          echo "http_cache_versions: $http_cache_versions"
+
+  nx-project-setup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Run Nx Project Setup
+        uses: mage-os/github-actions/nx-integration-tests-setup@main
+        with:
+          repository: ${{ inputs.repository }}
+          ref: ${{ inputs.pr_head }}
+          pr_base: ${{ inputs.pr_base }}
+      
+  integration-tests:
+    needs: 
+      - matrix-calculator
+      - nx-project-setup
+    strategy:
+      fail-fast: false
+      matrix:
+        php_version: ${{ fromJSON(needs.matrix-calculator.outputs.php_versions) }}
+        database_version: ${{ fromJSON(needs.matrix-calculator.outputs.database_versions) }}
+        search_version: ${{ fromJSON(needs.matrix-calculator.outputs.search_versions) }}
+        message_queue_version: ${{ fromJSON(needs.matrix-calculator.outputs.message_queue_versions) }}
+        cache_version: ${{ fromJSON(needs.matrix-calculator.outputs.cache_versions) }}
+        http_cache_version: ${{ fromJSON(needs.matrix-calculator.outputs.http_cache_versions) }}
+    runs-on: ubuntu-latest
+    steps:
+      # - name: Running With

```

Comment:
Do we need this? If not, can we remove it?

Reply:
just debug, have removed